tsdb/head.go

```diff
@@ -1542,6 +1542,10 @@ func (h *headIndexReader) Symbols() index.StringIter {
 // LabelValues returns the possible label values
 func (h *headIndexReader) LabelValues(name string) ([]string, error) {
 	h.head.symMtx.RLock()
+	if h.maxt < h.head.MinTime() || h.mint > h.head.MaxTime() {
```

Comment:
We should probably add a test for this behaviour here and for `LabelNames`. I'd put it in `head_test.go` now.

Reply:
added the test in this commit: https://github.com/prometheus/prometheus/pull/7288/commits/9c57a70e1adddbee5760fa82beface259be8122a, let me know what you think.  

Also @codesome, @krasi-georgiev would you mind reviewing this tsdb change?

Reply:
yes this change looks good.

Reply:
Small nit; should be consistent about whitespace before/after the if statement. Here you don't have a newline before but do after, and in `LabelNames` you have newline before and after. I would go with newline before and after personally.